### PR TITLE
Adds a new axe for Atmos-traitors

### PR DESCRIPTION
**Pull request by IcyV**
Give Atmos-traitors a unique 10TC axe to buy.

This axe starts with a single charge and can gain more charges over time by destroying windows and grilles. Each grille or window destroyed has a 4% chance of giving another charge to the axe. On a charged hit the axe will hit for a force of 30 and 30 burn. It will otherwise function as a normal fireaxe if out of charges.

nknwxga

Sprites courtesy of Fullofskittles
:cl: IcyV
add: Adds a unique axe for Atmos-traitors
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
I like the idea in concept, but the bonus damage seems a bit much.

**Comment by FlattestGuitar** [comment]
I like the sprite. Perhaps make it recharge with a lower damage though?

File: code/game/objects/items/weapons/twohanded.dm [hunk 1]
```diff
@@ -576,3 +576,48 @@
...
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	var/charged = 1
+	
+/obj/item/weapon/twohanded/energizedfireaxe/update_icon()
+	if(wielded)
+		icon_state = "fireaxe2"
+	else
+		icon_state = "fireaxe0"
+
+/obj/item/weapon/twohanded/energizedfireaxe/afterattack(atom/A as mob|obj|turf|area, mob/user as mob, proximity)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: this should beeeeee

`/obj/item/weapon/twohanded/energizedfireaxe/afterattack(atom/A, mob/user, proximity)`

File: code/game/objects/items/weapons/twohanded.dm [hunk 2]
```diff
@@ -576,3 +576,48 @@
...
+	if(!proximity) return
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: put the return on a new line, please.

File: code/game/objects/items/weapons/twohanded.dm [hunk 3]
```diff
@@ -576,3 +576,48 @@
...
+	if(wielded)
+		if(istype(A, /mob/living))
+			var/mob/living/Z = A
+			if(charged == 1)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: `if(charged)`

**Comment by IcyV** [comment]
Yeah, I'm not entirely sure still. 70 seems like a bit much but after slamming away on mobs for a while it doesn't seem overly crazy. It's a high cost for being able to do essentially 3 hits in one on a single mob.

File: code/game/objects/items/weapons/twohanded.dm [hunk 4]
```diff
@@ -576,3 +576,48 @@
...
+				charged = 0
+				Z.take_organ_damage(0,70)
+				user.visible_message("<span class='danger'>[user] slams the charged axe into [Z.name] with all their might!</span>")
+				playsound(src.loc, 'sound/magic/lightningbolt.ogg', 5, 1)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: doesn't need `src.` before the loc here.

File: code/game/objects/items/weapons/twohanded.dm (hunk 2 above)
Comment (IcyV) [inline code/game/objects/items/weapons/twohanded.dm]: DIdn't even notice I forgot to space that, my bad

File: code/game/objects/items/weapons/twohanded.dm [hunk 5]
```diff
@@ -576,3 +576,48 @@
...
+				var/datum/effect/system/spark_spread/sparks = new /datum/effect/system/spark_spread
+				sparks.set_up(1, 1, src)
+				sparks.start()
+				
+		if(A && wielded && (istype(A,/obj/structure/window) || istype(A,/obj/structure/grille)))
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: put spaces after the `A,` please.

**Comment by Fox-McCloud** [comment]
Just a few styling issues---aside from that, it looks good.

**Comment by IcyV** [comment]
Getting right on those, thank ya.

**Comment by Earthdivine** [comment]
100 damage on a first swing seems a bit much when it will take 2 after that to finish off the mob (yea I know it'll take 4 but the person attacked will be in crit).

Knocking it down to 40-50 bonus (70-80) might be better since then you can still kill them fairly well, but gives them time to react and run or fight. Plus it would only take 3 swings to knock them to crit from that.

**Comment by IcyV** [comment]
That's probably a bit more fair, yeah. 

I was also toying with the idea of letting it recharge over time somehow but at a much lower damage per charged hit.

File: code/game/objects/items/weapons/twohanded.dm [hunk 6]
```diff
@@ -576,3 +576,49 @@
...
+	
+/obj/item/weapon/twohanded/energizedfireaxe/update_icon()
+	if(wielded)
+		icon_state = "fireaxe2"
+	else
+		icon_state = "fireaxe0"
+
+/obj/item/weapon/twohanded/energizedfireaxe/afterattack(atom/A, mob/user, proximity)
+	if(!proximity) 
+	return
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: need to tab this in one.

Comment (IcyV) [inline code/game/objects/items/weapons/twohanded.dm]: Yeah, already got it done. My editor decided to undo my tab at some point and I didn't notice.

**Comment by IcyV** [comment]
Changed up how it works a little bit and updated the main post.

**Comment by Belario** [comment]
:+1: Broken windows will be the new meta.

**Comment by taukausanake** [comment]
The sprite keeps its charge icon even after being used on someone? Shouldn't it stop looking so shiny after loosing the charge? I'd do that personally so it's not found so quickly

**Comment by IcyV** [comment]
@taukausanake It's specifically while it's wielded. My take on it was that it's essentially always active, it just needs a certain level of charge to actually do anything. You can hide it as long as it's not wielded.

**Comment by Davdichan** [comment]
Does smashing a grill destroy one of the bars? Smash grill, grab bar reassemble and repeat till charge gained.

Given this is a high damage weapon, it should be a bit more obvious it's a traitor item even without a charge. Just my two cents

**Comment by IcyV** [comment]
@Davdichan 
>Does smashing a grill destroy one of the rods?

Correct, it works like the normal fireaxe in that regard. You won't get rods back. Windows will leave shards, but you'll have to burn quite a lot of welder fuel and time to get anywhere with that.

>I'd almost prefer it act as a mini-power sink that needs to you hack a power using object to gain a charge, like an APC, electrified grill or exposed wire.

I considered that for a while and played with it a bit but it just didn't feel all that great. It was way too easy to hide in a corner or just charge on the spot virtually anywhere. Having to smash your way to a charge means you're going to be super loud and very obvious in your intent, especially with the sprite.

>Given this is a high damage weapon, it should be a bit more obvious it's a traitor item even without a charge.

Also considered that, but then you have to recall that there are two axes on average on the station ever. It's super easy to check and see if either of those are there or missing. Using the axe will be ultra blatant, but you might get just a little bit more time carrying it around if people don't instantly know. It gives you a chance to walk around for just a bit before people get suspicious of why you have an axe if, say, the atmos one is still there.

**Comment by Fox-McCloud** [comment]
Oof. Merge conflicts @IcyV 

Probably sprites

**Comment by IcyV** [comment]
@Fox-McCloud 

The faintest idea at best. I'll get working on that when I get off work here later though.

**Comment by Fox-McCloud** [comment]
Generally speaking sprite conflicts are a pain to solve. My best recommendation is to merge from master, then out and out delete your weapons.dmi and your lefthand and righthand dmi's.

After that, redownload those files from master nad put them where they belong---then manuallly re-adde the axe in-hands and object.

**Comment by IcyV** [comment]
@Fox-McCloud I can only assume I did everything horribly wrong. I'll be a man and blame the weather.

**Comment by Fox-McCloud** [comment]
Were there more conflicts than just the sprite files?

**Comment by IcyV** [comment]
Nothing showed up for me, so I'm not entirely sure.

**Comment by Fox-McCloud** [comment]
Well uh. Hm. How exactly did you resolve the conflicts?

I'll go over my process again, as it was probably about as clear as mud.

download these files:
https://github.com/ParadiseSS13/Paradise/blob/master/icons/obj/weapons.dmi
https://github.com/ParadiseSS13/Paradise/blob/master/icons/mob/inhands/items_lefthand.dmi
https://github.com/ParadiseSS13/Paradise/blob/master/icons/mob/inhands/items_righthand.dmi

and overwrite the files on this fire-axe branch with them, completely. Once you've done that, then go in and re-add the fire axe sprites, then push the changes. That *should* resolve the conflicts---if it doesn't...well...I'm really not sure at that stage.

**Comment by IcyV** [comment]
Yep, that's exactly what I did. Pushed it and then apparently nothing.

**Comment by Fox-McCloud** [comment]
Weird. Well. You can hard-reset your branch and start over, orrrrr you can just make an entirely new PR with this very one--if you go with the latter, it wouldn't have to wait 24 hours provided it was identical to this one, as this PR is merge-ready (minus conflicts).

**Pull request closed without merging**